pkg/common/cns-lib/node/nodes.go

```diff
@@ -122,11 +121,28 @@ func (nodes *Nodes) csiNodeAdd(obj interface{}) {
 	nodeName := csiNode.Name
 	nodeUUID := k8s.GetNodeIdFromCSINode(csiNode)
 	if nodeUUID == "" {
-		log.Warnf("csiNodeAdd: nodeId is either empty. CSINode object: %v",
-			csiNode)
+		log.Warnf("csiNodeAdd: nodeId is either empty. CSINode object: %v", csiNode)
 		return
 	}
-	err := nodes.cnsNodeManager.RegisterNode(ctx, nodeUUID, nodeName)
```

Comment:
Could you also add an explanation here to say why we need to call DiscoverNode here as RegisterNode already calls DiscoverNode function?

Reply:
fixed it. We are now calling RegisterNode twice, one with node ID obtained from CSINode object, and if that failed we call RegisterNode again with providerID obtained from the Node.